File: Examen1/2024_2/EventMasterSoft/EventMasterSoftPersistance/DAO/IDAO.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace EventMasterSoftPersistance.DAO
{
    public interface IDAO <T>
    {
        int Insertar(T objeto);
        int Modificar(T objeto);
        int Eliminar(T objeto);
        T BuscarPorId(int id);
        List<T> ListarTodos();
    }
}

File: Lab11y14/Lab11-Prog3-2026-1/Backend/pokesoft_solution/pokesoft_persistence/src/pe/edu/pucp/pokesoft/persistance/dao/IDAO.java
package pe.edu.pucp.pokesoft.persistance.dao;

import java.util.List;

public interface IDAO <T> {
    int insertar(T objeto);
    int modificar(T objeto);
    int eliminar(int id);
    T buscarPorId(int id);
    List<T> listarTodos();
}

